ci/scripts/check_bazel_target_names.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import string
+import subprocess
+import sys
+from typing import Callable, Iterable, List, Set, Tuple
+
+
+class BazelQueryRunner:
+
+    def __init__(self, backend: Callable[[str], List[str]] = None):
+        self._backend = backend
+
+    def find_targets_with_banned_chars(self) -> Iterable[Tuple[str, Set[str]]]:
+        """Find targets in //... with names containing disallowed characters.
+
+        Bazel allows a liberal set of characters in target names [0], which
+        enables type confusion bugs to go unnoticed. For example, a tuple's
+        string representation, complete with parentheses and commas, can
+        accidentally be inserted into a target name via `string.format()`.
+
+        [0]: https://bazel.build/concepts/labels#target-names
+
+        Yields:
+          A tuple for each target that has banned characters in its name. The
+          first element is the Bazel target's name. The second element is the
+          set of characters that must be removed from the target's name.
+
+        """
+        allowed_chars = set(string.ascii_letters + string.digits + '/:_-.')
+        for target in self.query("//..."):
+            if bad_chars := set(target) - allowed_chars:
+                yield (target, bad_chars)
+
+    def find_empty_test_suites(self) -> Iterable[str]:
```

Comment:
Does it seem weird to define `find_empty_test_suites` in this file? I think it belongs because incorrectly-assembled target names are a likely cause of empty test suites.

Reply:
Malformed test suites is also addressed in "check_bazel_tags.sh"

We should probably test for that separately to help people understand why we're being picky about test_suites.

Reply:
I did a little reorganizing.

Now we have `check_bazel_target_names.py` and `check_bazel_test_suites.py`. I moved the check for non-manual test suites into the latter script. This enabled me to write some unit tests — in particular, I'm happy to have some coverage on the behavior of the funky regex for tag matching.

WDYT?